### PR TITLE
[SPARK-42297][SQL] Assign name to _LEGACY_ERROR_TEMP_2412

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to assign name to _LEGACY_ERROR_TEMP_2412, "UNSUPPORTED_EXPR_FOR_WINDOW".

### Why are the changes needed?

We should assign proper name to _LEGACY_ERROR_TEMP_*

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

`./build/sbt "sql/testOnly org.apache.spark.sql.SQLQueryTestSuite*"`

### CODE REVIEW

**Comment by itholic** [comment]
cc @srielau @MaxGekk @cloud-fan PTAL when you find some time 🙏

**Review by srielau — APPROVED** [review]

**Review by LuciferYang — APPROVED** [review]

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala [hunk 1]
```diff
@@ -771,6 +771,31 @@ class QueryExecutionErrorsSuite
     assert(e.getErrorClass === "STREAM_FAILED")
     assert(e.getCause.isInstanceOf[NullPointerException])
   }
+
+  test("UNSUPPORTED_EXPR_FOR_WINDOW: to_date is not supported with WINDOW") {
```
Comment (khalidmammadov) [inline sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala:775]: Does it also need JIRA number here?

Comment (MaxGekk) [reply sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala:775]: It is optional. Usually, we add an JIRA number to refer a bug report that the test addresses. JIRA numbers don't bring additional info for regular tests like this.

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master/3.4.
Thank you, @itholic and @srielau @khalidmammadov @LuciferYang for review.

**Pull request closed without merging**